File: ruby/grains/grains.rb
class Grains
  BOARD_SIZE = 64
  def self.square(location)
    raise ArgumentError unless (1..BOARD_SIZE).include?(location)
    1 << (location - 1)
  end

  def self.total
    square(BOARD_SIZE) * 2 - 1
  end
end

File: c/grains/src/grains.c
#include "grains.h"

unsigned long long square(unsigned int field) {
  return ((field > 0) && (field <= 64)) ? 1ULL << (field - 1) : 0;
}

unsigned long long total() {
  unsigned int i;
  unsigned long long sum = 0;
  for (i = 1; i <= 64; i++)
    sum += 1ULL << i;
  return sum;
}
